catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsTest.java

```diff
@@ -0,0 +1,500 @@
+        when(resourceRequestMock.getAttributeValue()).thenReturn(testUri);
+        when(resourceRequestMock.getAttributeName()).thenReturn("resource-uri");
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+
+        verify(resourceRequestMock).getProperties();
+    }
+
+    @Test(expected = ResourceNotSupportedException.class)
+    public void testGetResourceCatchesStopProcessingException()
+            throws ResourceNotFoundException, IOException, ResourceNotSupportedException {
+        boolean isEnterprise = true;
+        boolean fanoutEnabled = true;
+        String resourceName = "bob";
+
+        when(resourceRequestMock.getProperties()).thenThrow(StopProcessingException.class);
+        when(resourceRequestMock.getAttributeValue()).thenReturn(testUri);
+        when(resourceRequestMock.getAttributeName()).thenReturn("resource-uri");
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+
+        verify(resourceRequestMock).getProperties();
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceNullSourceNameNotEnterpriseThrowsResourceNotFoundException()
+            throws Exception {
+
+        boolean isEnterprise = false;
+        boolean fanoutEnabled = false;
+        String resourceName = null;
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+    }
+
+    @Test(expected = ResourceNotSupportedException.class)
+    public void testGetResourceInfoThrowsResourceNotSupportedException() throws Exception {
+        boolean isEnterprise = true;
+        boolean fanoutEnabled = true;
+        String resourceName = "bob";
+
+        when(resourceRequestMock.getAttributeValue()).thenReturn("bob");
+        when(resourceRequestMock.getAttributeName()).thenReturn("bob");
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceInfoCatchUnsupportedQueryExceptionOrFederationException()
+            throws Exception {
+        boolean isEnterprise = true;
+        boolean fanoutEnabled = true;
+        String resourceName = "bob";
+
+        when(resourceRequestMock.getAttributeValue()).thenReturn(testUri);
+
+        when(resourceRequestMock.getAttributeName()).thenReturn("resource-uri");
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetResourceInfoUriNotBlank() throws Exception {
```

Comment:
I might be missing something but this test looks the exact same as the one above.